lib/internal/streams/writable.js

```diff
@@ -259,11 +258,17 @@ function WritableState(options, stream, isDuplex) {
 function resetBuffer(state) {
   state.buffered = [];
   state.bufferedIndex = 0;
-  state.state |= kAllBuffers | kAllNoop;
+  state.state |= kAllBuffers | kSimpleBuffer;
 }
 
 WritableState.prototype.getBuffer = function getBuffer() {
-  return ArrayPrototypeSlice(this.buffered, this.bufferedIndex);
+  return (this.state & kSimpleBuffer) === 0
+    ? this.buffered.slice(this.bufferedIndex)
+    : this.buffered.map(chunk => ({
```

Comment:
If this is a hot path, I think it would be faster to avoid the functional array methods and use a normal loop instead. Same for the other use of `map()`.

Reply:
IMO This is not a hot path.